django/contrib/postgres/search.py

```diff
@@ -60,6 +60,13 @@ def __init__(self, *expressions, **extra):
 
     def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
         resolved = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        resolved.set_source_expressions([
+            Coalesce(
+                expression if isinstance(expression.output_field, (CharField, TextField))
+                else Cast(expression, CharField()),
```

Comment:
Just a thought... Do we want to `::varchar` or `::text`? Both use the same underlying data type for storage, so `varchar` is no faster, and most of PostgreSQL's text manipulation functions will return `text` regardless of string types input.

Reply:
I don't have any preference really, both are using `varlena` as you said. I could switch to `TextField` if you think it's more idiomatic.